tools/FancyZones_WindowStyles/FancyZones_WindowStyles.cpp

```diff
@@ -0,0 +1,225 @@
+// FancyZones_WindowStyles.cpp : Defines the entry point for the application.
+//
+
+#include "pch.h"
+#include "FancyZones_WindowStyles.h"
+
+#include <shlobj.h>
+
+#include <filesystem>
+#include <fstream>
+
+inline std::optional<std::wstring> get_last_error_message(const DWORD dw)
+{
+    std::optional<std::wstring> message;
+    try
+    {
+        const auto msg = std::system_category().message(dw);
+        message.emplace(begin(msg), end(msg));
+    }
+    catch (...)
+    {
+    }
+    return message;
+}
+
+inline std::wstring get_last_error_or_default(const DWORD dw)
+{
+    auto message = get_last_error_message(dw);
+    return message.has_value() ? message.value() : L"";
+}
+
+std::filesystem::path get_desktop_path()
+{
+    wchar_t* p;
+    if (S_OK != SHGetKnownFolderPath(FOLDERID_Desktop, 0, NULL, &p)) return "";
+
+    std::filesystem::path result = p;
+    CoTaskMemFree(p);
+
+    return result;
+}
+
+class Logger
+{
+private:
+    inline static std::wofstream logger;
+
+public:
+    ~Logger()
+    {
+        logger.close();
+    }
+
+    static void init(std::string loggerName)
+    {
+        std::filesystem::path rootFolder(get_desktop_path());
+
+        auto logsPath = rootFolder;
+        logsPath.append(L"window_styles.txt");
+
+        logger.open(logsPath.string(), std::ios_base::out | std::ios_base::app);
+    }
+
+    template<typename FormatString, typename... Args>
+    static void log(const FormatString& fmt, const Args&... args)
+    {
+        logger << std::format(fmt, args...) << std::endl;
+    }
+};
+
+void LogInfo(HWND window)
+{
+    auto style = GetWindowLong(window, GWL_STYLE);
+    auto exStyle = GetWindowLong(window, GWL_EXSTYLE);
+
+    Logger::log(L"Style: WS_BORDER           {}", ((style & WS_BORDER) == WS_BORDER));
```

Comment:
Let's log window handle, name and process path first to identify the window in the log file.

Reply:
Btw. process name can be helpful (e.g. `Calculator.exe` vs `ApplicationFrameHost.exe`).

Reply:
I thought about it, but wouldn't it be kind of personal info? We're removing app names in the `app-zone-history.json`.

Reply:
But how do you identify different windows in the same file then? This is a difficult point. 🤔

We could log hwnd and process name/path. That might help for analysation. The name isn't a must have for debugging and can be an optional setting in the ux maybe.